pddl/core.py

```diff
@@ -41,7 +41,7 @@ def __init__(
         self,
         name: namelike,
         requirements: Optional[Collection["Requirements"]] = None,
-        types: Optional[Dict[namelike, Collection[namelike]]] = None,
+        types: Optional[Dict[namelike, Optional[namelike]]] = None,
```

Comment:
idea: we could have a higher-level data structure instead of raw dictionaries, that would make easier to ask queries, e.g. all the parents of a given type, and guarantee other invariants, e.g. absence of cycles in the inheritance graph.

Reply:
@haz  and I discussed that in #70 and we agreed on using a dictionary for the moment. However, as actually seen in this PR, to accommodate things specific parts of the code are becoming increasingly fuzzy. So, I guess that ultimately your proposal will be the way to go to maintain readability.